protocol/bc/merkle.go

```diff
@@ -14,7 +13,7 @@ var (
 
 // CalcMerkleRoot creates a merkle tree from a slice of transactions
 // and returns the root hash of the tree.
-func CalcMerkleRoot(transactions []*bc.Tx) (root bc.Hash, err error) {
+func CalcMerkleRoot(transactions []*Tx) (root Hash, err error) {
```

Comment:
Would you be ok with renaming this to MerkleRoot? I'd prefer to name a function after what it returns, rather than what it does.